nslsii/temperature_controllers/lakeshore.py

```diff
@@ -0,0 +1,126 @@
+from collections import OrderedDict
+from ophyd import (Component, DynamicDeviceComponent, Device, EpicsSignal,
+                   EpicsSignalRO, PVPositioner)
+
+
+class Lakeshore336(Device):
+    '''An ``ophyd.Device`` class to be used for Lakeshore336 temp controllers.
+
+    This is a class for controlling Lakeshore model 336 temperature controllers
+    The defaults are set to include four sensors (A, B, C and D) and four
+    control loops (1,2,3 and 4) however this is customizable at initialization
+    time via the kwargs ``temperatures`` and ``controls`` respectively. To
+    instantiate the default settings use the command:
+    ``my_controller = Lakeshore336(PVprefix, name='my_controller')``
+
+    Parameters
+    ----------
+    *args: list
+        The arguments that are passed to ``ophyd.Device``.
+    temperatures: list
+        A list of temperature sensor 'channels' to instantiate, the default is
+        ['A','B','C','D'].
+    controls: list
+        A list of heater control 'channels' to instantiate, the default is
+        [1,2,3,4].
+    **kwargs: dict
+        The keyword arguments passed to ``ophyd.Device``.
+    '''
+
+    def _set_fields(fields, device, prefix, **kwargs):
```

Comment:
While not wrong, this naming is very confusing.  When passing in a class (that is an instance of `type`) you usually name it `cls` or `klass` (but not `class` because that is a reserved keyword) to be clear to the user that this is not an _instance_ of the class, but the class it's self.

Reply:
This also won't work as python will interpret this as a method, bind it to the instance and when you use it as `__init__` I think that this will raise a type error because you passed it the wrong number of positional args.

You ether need to add a `@staticmethod` decorator or just move it out of the class.

Reply:
I have changed  `device` to `cls` as suggested, see comment above about moving this from __init__.

Reply:
have made this a factory function which solves this issue